prboom2/src/g_game.c

```diff
@@ -3179,13 +3182,74 @@ void G_BeginRecording (void)
 {
   int i;
   byte *demostart, *demo_p;
+  int num_extensions = 0;
   demostart = demo_p = malloc(1000);
   longtics = 0;
 
-  if (umapinfo_loaded)
+  // ano - jun2019 - add the extension format if needed
+  if (gamemapinfo)
+  {
+    num_extensions++;
+  }
+
+  if (num_extensions > 0)
   {
-	  *demo_p++ = 255;
+    // demover
+    *demo_p++ = 0xFF;
+    // signature
+    *demo_p++ = prdemosig[0]; // 'P'
+    *demo_p++ = prdemosig[1]; // 'R'
+    *demo_p++ = prdemosig[2]; // '+'
+    *demo_p++ = prdemosig[3]; // 'U'
+    *demo_p++ = prdemosig[4]; // 'M'
+    *demo_p++ = '\0';
+    // extension version
+    *demo_p++ = 1;
+    //
+    *demo_p++ =  num_extensions & 0xff;
+    *demo_p++ = (num_extensions >> 8) & 0xff;
+
+    if (gamemapinfo)
+    {
+      int mapname_len = strnlen(gamemapinfo->mapname, 9);
+
+      // ano - note that the format has each length by each string
+      // as opposed to a table of lengths
+      *demo_p++ = 0x08;
+      *demo_p++ = 'U';
+      *demo_p++ = 'M';
+      *demo_p++ = 'A';
+      *demo_p++ = 'P';
+      *demo_p++ = 'I';
+      *demo_p++ = 'N';
+      *demo_p++ = 'F';
+      *demo_p++ = 'O';
```

Comment:
Why do we have to embed the entire "UMAPINFO" string here? And whats the matter with `0x08`? Can't we just enumerate all possible features and assign them simple numerical values?

Reply:
100% certainty there will be no clashes with anything else? 8 bytes a demo isnt going to cause any real issues, is it?

Reply:
Ah, OK, so it's meant to indicate the length of the identifier string that follows. I didn't get that on first parsing.